clippy_lints/src/methods/mod.rs

```diff
@@ -1532,6 +1555,9 @@ impl<'tcx> LateLintPass<'tcx> for Methods {
                     if match_def_path(cx, fn_def_id, &paths::PUSH_STR) {
                         lint_single_char_push_string(cx, expr, args);
                     }
+                    if match_def_path(cx, fn_def_id, &paths::INSERT_STR) {
```

Comment:
Why do `push_str` and `insert_str` handle separately from other methods?  It seems that it works great inside the big `match method_names.as_slice()` above.

Reply:
for insert_str, I blindly copied most of the push_str code, but yeah it seemed a bit strange where the code was.

Reply:
It would seem both fit inside the big `match` indeed, but that part of the code is using `utils::method_calls`, which here:
https://github.com/rust-lang/rust-clippy/blob/d88b9b7543756afef81dcb2e2c37a11603d513d5/clippy_lints/src/utils/mod.rs#L390-L392

Is excluding arguments that come from expansion, which includes the `Self` argument. In this PR we added a case where a macro is used in that position:

```rust
    get_string!().insert_str(1, "?");
```

If we want to keep linting in that case, we can't move these lints inside the big `match`.